pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1544,6 +1544,72 @@ func PublishContentInfoToZedCloud(ctx *zedagentContext, uuid string,
 	}
 }
 
+// PublishVolumeToZedCloud is called per change, hence needs to try over all management ports
+// When volume status is nil it means a delete and we send a message
+// containing only the UUID to inform zedcloud about the delete.
+func PublishVolumeToZedCloud(ctx *zedagentContext, uuid string,
+	volStatus *types.VolumeStatus, iteration int) {
+
+	log.Infof("PublishVolumeToZedCloud uuid %s", uuid)
+	var ReportInfo = &info.ZInfoMsg{}
+
+	volumeType := new(info.ZInfoTypes)
+	*volumeType = info.ZInfoTypes_ZiVolume
+	ReportInfo.Ztype = *volumeType
+	ReportInfo.DevId = *proto.String(zcdevUUID.String())
+	ReportInfo.AtTimeStamp = ptypes.TimestampNow()
+
+	ReportVolumeInfo := new(info.ZInfoVolume)
+
+	ReportVolumeInfo.Uuid = uuid
+	ReportVolumeInfo.State = info.ZSwState_HALTED
```

Comment:
HALTED might not make sense for a volume which is deleted. INIT makes more sense I think.

Reply:
Makes sense. I will update it.

Reply:
Resolved.